core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/solver/recaller/BestSolutionRecaller.java

```diff
@@ -118,22 +118,26 @@ public void processWorkingSolutionDuringMove(Score score, AbstractStepScope<Solu
         }
         if (bestScoreImproved) {
             phaseScope.setBestSolutionStepIndex(stepScope.getStepIndex());
-            Solution_ newBestSolution = solverScope.getScoreDirector().cloneWorkingSolution();
-            updateBestSolutionAndFire(solverScope, score, newBestSolution);
+            updateBestSolutionAndFire(solverScope);
         } else if (assertBestScoreIsUnmodified) {
             solverScope.assertScoreFromScratch(solverScope.getBestSolution());
         }
     }
 
     public void updateBestSolutionAndFire(SolverScope<Solution_> solverScope) {
-        Solution_ newBestSolution = solverScope.getScoreDirector().cloneWorkingSolution();
-        Score newBestScore = solverScope.getSolutionDescriptor().getScore(newBestSolution);
-        updateBestSolutionAndFire(solverScope, newBestScore, newBestSolution);
+        updateBestSolutionWithoutFiring(solverScope);
```

Comment:
This was a bug previously introduced and not noticed.
Firing BKS should not have happened here, and now it doesn't again.

Reply:
Maybe needs a test.